source/data-modeling/associations.txt

```diff
@@ -515,3 +515,410 @@ method to remove all embedded ``Album`` documents from the ``Band`` class:
+   :language: ruby
+   :start-after: # start-custom-inverse-keys
+   :end-before: # end-custom-inverse-keys
+   :emphasize-lines: 9, 20
+
+Custom Scopes
+~~~~~~~~~~~~~
+
+You can set a specific scope on an association by using the ``scope`` parameter.
+The ``scope`` parameter determines which documents {+odm+} considers to be part
+of an association. A scoped association returns only documents that match the
+scope conditions when queried. You can set the ``scope`` to either a ``Proc`` with an arity
+of zero or a ``Symbol`` that references a named scope on the associated model.
+The following example sets custom scopes on associations in a ``Band`` class:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-custom-scope
+   :end-before: # end-custom-scope
+
+.. note::
+
+   You can add documents that do not match the scope conditions to an
+   association. {+odm+} saves the documents to the database and they will appear
+   in associated memory, however, the documents are not present when the
+   association is queried in the future.
+
+Validations
+~~~~~~~~~~~
+
+By default, {+odm+} validates that the children of any associations are present
+when it loads the association into memory by using the ``validates_associated``
+macro. This applies to the following association types:
+
+- ``embeds_many``
+- ``embeds_one``
+- ``has_many``
+- ``has_one``
+- ``has_and_belongs_to_many``
+
+You can turn off this validation behavior by setting the ``validate`` macro to
+``false`` when defining the association, as shown in the following example:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-validation-false
+   :end-before: # end-validation-false
+
+Polymorphism
+~~~~~~~~~~~~
+
+{+odm+} supports polymorphism on the child classes of one-to-one and one-to-many associations.
+Polymorphic associations allows a single association to contain objects of different class
+types. You can define a polymorphic association by setting the ``polymorphic``
+option to ``true`` in a child association and adding the ``as`` option to the
+parent association. The following example creates a polymorphic association in a
+``Band`` class:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-polymorphic
+   :end-before: # end-polymorphic
+
+In the preceding example, the ``:featured`` association in the ``Band`` class can contain either a
+``Label`` or ``Album`` document.
+
+.. important::
+
+   {+odm+} supports polymorphism only from child to parent. You cannot specify
+   a parent ``has_one`` or ``has_many`` association as polymorphic.
+
+``has_and_belongs_to_many`` associations do not support polymorphism.
+
+Custom Polymorphic Types
+````````````````````````
+
+Starting in version 9.0.2, {+odm+} adds support for custom polymorphic types through
```

Comment:
s:

```suggestion
Starting in version 9.0.2, {+odm+} supports custom polymorphic types through
```

is the support for polymorphic types new and the global registry is what makes it work? or have polymorphic types always been supported and the global registry piece is new?